articles/openshift/oshift-how-add-domains-proxy-whitelist.md

```diff
@@ -0,0 +1,122 @@
+oc policy add-role-to-user edit-proxy-whitelist $username -n whitelist --role-namespace='whitelist'
+```
+### Modifying the whitelist
+
+There are two tools which can be used to modify the `proxy-whitelist` Config Map. As previously stated you will need to have the necessary rights to edit this object within the `whitelist` project. 
+
+#### Web console
+
+1. To view the `proxy-whitelist` Config Map within the `whitelist` project, navigate to the following URL (ensuring you replace \<your-deployment-name>): `https://ocp.<your-deployment-name>.cna.ukcloud.com:8443/console/project/whitelist/browse/config-maps/proxy-whitelist`
+![whitelist ConfigMaps](images/oshift-proxy-configmap.png)
+
+>[!TIP]
+>To whitelist all subdomains, precede the domain with a `.` e.g. `.domain.com`
+
+2. Click `Actions` followed by `Edit` to modify the values within the `proxy-whitelist.txt` key. Each domain should be on its own line:
+![Edit proxy-whitelist](images/oshift-proxy-editproxywhitelist.png)
+
+3. Click `Save` when finished
+
+#### oc client
+>[!NOTE]
+>To authenticate using the `oc` client you will need an API token. Retrieve this by logging in to the Web Console and clicking your name in the top right; choosing `Copy Login Command`. This can then be pasted from your buffer into a terminal where you have the `oc` client installed.
+
+1. Change to the `whitelist` project: 
+   
+   ```bash
+   $ oc project whitelist
+   ```
+2. Edit the `proxy-whitelist` Config Map:
+
+   ```bash
+   $ oc edit cm proxy-whitelist
+   ```
+
+   This will display the object definition in yaml format in your default text editor:
+
+   ```
+   # Please edit the object below. Lines beginning with a '#' will be ignored,
+   # and an empty file will abort the edit. If an error occurs while saving this file will be
+   # reopened with the relevant failures.
+   #
+   apiVersion: v1
+   data:
+     proxy-whitelist.txt: ""
+   kind: ConfigMap
+   metadata:
+     creationTimestamp: 2019-01-18T01:18:44Z
+     name: proxy-whitelist
+     namespace: whitelist
+     resourceVersion: "5385199"
+     selfLink: /api/v1/namespaces/whitelist/configmaps/proxy-whitelist
+     uid: ff7df4a9-1abe-11e9-965a-fa163ecc60e0
+   ```
+
+   Within the object definition data attribute is the `proxy-whitelist.txt` key. By default this will be an empty single line string but you can modify this to be a multi-line string by changing `""` to `|-` as below and indenting the domains you wish to add with two spaces. For example:
+
+   ```
+   proxy-whitelist.txt: |-
+     firstdomain.com
+     .seconddomain.com
+   ```
+
+   Save your changes to persist to the Config Map, otherwise cancel your edit to not make any changes.
+
+## Further reading
+
+https://wiki.squid-cache.org/SquidFaq/SquidAcl
+
+## Feedback
+
+If you have any comments on this document or any other aspect of your UKCloud experience, send them to <products@ukcloud.com>.
```

Comment:
Update this to the latest feedback information:

If you find an issue with this article, click **Improve this Doc** to suggest a change. If you have an idea for how we could improve any of our services, visit [UKCloud Ideas](https://ideas.ukcloud.com). Alternatively, you can contact us at <products@ukcloud.com>.